tooling-support-tests/src/test/java/org/mule/runtime/module/tooling/MetadataTypesTestCase.java

```diff
@@ -281,5 +282,33 @@ public void internalErrorInsideResolver() {
     }
   }
 
+  @Test
+  public void metadataCache() {
+    OperationElementDeclaration operationElementDeclaration = resolverUsesResourcesCacheOP(CONFIG_NAME, "KEY_1");
+    // We need to clean up cache first...
+    session.disposeMetadataCache(operationElementDeclaration);
+
+    // First call
+    validateMetadataCacheResolve(operationElementDeclaration, "KEY_1");
+
+    // Second call
+    validateMetadataCacheResolve(resolverUsesResourcesCacheOP(CONFIG_NAME, "KEY_2"), "KEY_1 KEY_2");
```

Comment:
How do you know that is using caching information?

Reply:
The resolver caches the input key and keeps appending keys to the cache and returns the cached value.